File: frontend/src/types/display.ts
import type { Calendar } from './calendar';
import type { UserBalance } from './user';
import type { ChoreWithUser } from './chore';

export interface DisplayToken {
  id: string;
  name: string;
  token: string;
  created_at: string;
}

export interface CreateTokenInput {
  name: string;
}

export interface DisplayData {
  weather: Record<string, unknown> | null;
  calendars: Calendar[];
  allowances: UserBalance[];
  chores: ChoreWithUser[];
  background_url: string | null;
}

File: backend/src/models/display.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;
use uuid::Uuid;
use crate::models::{user::UserBalance, calendar::CalendarPublic, chore::ChoreWithUser};

#[derive(Debug, Serialize, Deserialize, FromRow)]
pub struct DisplayToken {
    pub id: Uuid,
    pub name: String,
    pub token: String,
    pub created_at: chrono::DateTime<chrono::Utc>,
}

#[derive(Debug, Deserialize)]
pub struct CreateTokenSchema {
    pub name: String,
}

#[derive(Debug, Serialize)]
pub struct DisplayData {
    pub weather: Option<serde_json::Value>,
    pub calendars: Vec<CalendarPublic>,
    pub allowances: Vec<UserBalance>,
    pub chores: Vec<ChoreWithUser>,
    pub background_url: Option<String>,
}